File: 프로그래머스/0/181854. 배열의 길이에 따라 다른 연산하기/배열의 길이에 따라 다른 연산하기.py
def solution(arr, n):
    L = len(arr)
    is_odd_length = L % 2 != 0

    return [
        val + n
        if (is_odd_length and i % 2 == 0) or (not is_odd_length and i % 2 != 0)
        else val
        for i, val in enumerate(arr)
    ];

File: 프로그래머스/0/181854. 배열의 길이에 따라 다른 연산하기/배열의 길이에 따라 다른 연산하기.js
function solution(arr, n) {
    const L = arr.length;
    const isOddLength = L % 2 !== 0;

    return arr.map((val, i) => {
        const condition = (isOddLength && i % 2 == 0) || (!isOddLength && i % 2 != 0);
        return condition ? val + n : val;
    });
}